sdk/database/dbplugin/v5/plugin_client_test.go

```diff
@@ -0,0 +1,146 @@
+	"github.com/hashicorp/vault/sdk/database/dbplugin/v5/proto"
+	"github.com/hashicorp/vault/sdk/helper/consts"
+	"github.com/hashicorp/vault/sdk/helper/pluginutil"
+	"github.com/hashicorp/vault/sdk/helper/wrapping"
+	"github.com/stretchr/testify/mock"
+	"google.golang.org/grpc"
+)
+
+func TestNewPluginClient(t *testing.T) {
+	type testCase struct {
+		config       pluginutil.PluginClientConfig
+		pluginClient pluginutil.PluginClient
+		expectedResp *DatabasePluginClient
+		expectedErr  error
+	}
+
+	tests := map[string]testCase{
+		"happy path": {
+			config: testPluginClientConfig(t),
+			pluginClient: &fakePluginClient{
+				connResp:     nil,
+				dispenseResp: gRPCClient{client: fakeClient{}},
+				dispenseErr:  nil,
+			},
+			expectedResp: &DatabasePluginClient{
+				client: &fakePluginClient{
+					connResp:     nil,
+					dispenseResp: gRPCClient{client: fakeClient{}},
+					dispenseErr:  nil,
+				},
+				Database: gRPCClient{proto.NewDatabaseClient(nil), context.Context(nil)},
+			},
+			expectedErr: nil,
+		},
+		"dispense error": {
+			config: testPluginClientConfig(t),
+			pluginClient: &fakePluginClient{
+				connResp:     nil,
+				dispenseResp: gRPCClient{},
+				dispenseErr:  errors.New("dispense error"),
+			},
+			expectedResp: nil,
+			expectedErr:  errors.New("dispense error"),
+		},
+		"error unsupported client type": {
+			config: testPluginClientConfig(t),
+			pluginClient: &fakePluginClient{
+				connResp:     nil,
+				dispenseResp: nil,
+				dispenseErr:  nil,
+			},
+			expectedResp: nil,
+			expectedErr:  errors.New("unsupported client type"),
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctx := context.Background()
+
+			mockWrapper := new(mockRunnerUtil)
+			mockWrapper.On("NewPluginClient", ctx, mock.Anything).
+				Return(test.pluginClient, nil)
+			defer mockWrapper.AssertNumberOfCalls(t, "NewPluginClient", 1)
+
+			resp, err := NewPluginClient(ctx, mockWrapper, test.config)
+			if test.expectedErr != nil && err == nil {
+				t.Fatalf("err expected, got nil")
+			}
+			if test.expectedErr == nil && err != nil {
+				t.Fatalf("no error expected, got: %s", err)
+			}
+			if test.expectedErr == nil && !reflect.DeepEqual(resp, test.expectedResp) {
+				t.Fatalf("Actual response: %#v\nExpected response: %#v", resp, test.expectedResp)
+			}
+		})
+	}
+}
+
+func testPluginClientConfig(t *testing.T) pluginutil.PluginClientConfig {
+	return pluginutil.PluginClientConfig{
+		Name:            "test-plugin",
+		PluginSets:      PluginSets,
+		PluginType:      consts.PluginTypeDatabase,
+		HandshakeConfig: HandshakeConfig,
+		Logger:          log.NewNullLogger(),
+		IsMetadataMode:  true,
+		AutoMTLS:        true,
+	}
+}
+
+var _ pluginutil.PluginClient = &fakePluginClient{}
+
+type fakePluginClient struct {
+	connResp grpc.ClientConnInterface
+
+	dispenseResp interface{}
+	dispenseErr  error
+}
+
+func (f *fakePluginClient) Conn() grpc.ClientConnInterface {
+	return f.connResp
```

Comment:
It looks like we don't actually set `connResp` to a non-nil value, is the field needed for testing or is it okay to just return nil?